cmd/sops/rotate.go

```diff
@@ -69,5 +70,8 @@ func rotate(opts rotateOpts) ([]byte, error) {
 	if err != nil {
 		return nil, common.NewExitError(fmt.Sprintf("Could not marshal tree: %s", err), codes.ErrorDumpingTree)
 	}
+	audit.SubmitEvent(audit.RotateEvent{
+		File: tree.FilePath,
+	})
```

Comment:
recording an audit event should happen before the operation is performed, not after. Otherwise a malicious user can simply block the DB connection and still use sops successfully.